### PR TITLE
feat: load profile-welcome block asynchronously without blocking

### PR DESCRIPTION
**Pull request by nitin-rachabathuni**
Please always provide the Jira Issue your PR is for, as well as test URLs where your change can be observed (before and after):

Jira ID: https://jira.corp.adobe.com/browse/EXLM-2364

Test URLs:

- Before: https://main--exlm--adobe-experience-league.hlx.live
- After: https://EXLM-2364--exlm--adobe-experience-league.hlx.live

### CODE REVIEW

File: blocks/profile-welcome/profile-welcome.js [hunk 1]
```diff
@@ -37,27 +37,45 @@ async function decorateProfileWelcomeBlock(block) {
...
-
-  if (isSignedIn) {
-    [profileData, ppsProfileData] = await Promise.all([
-      defaultProfileClient.getMergedProfile(),
-      defaultProfileClient.getPPSProfile(),
-    ]);
-  }
+  // eslint-disable-next-line no-async-promise-executor
+  const profilePromise = new Promise(async (resolve) => {
+    const isSignedIn = await isSignedInPromise;
```
Comment (nitin-rachabathuni) [inline blocks/profile-welcome/profile-welcome.js:44]: use isSignedInUser();

File: blocks/profile-welcome/profile-welcome.css [hunk 2]
```diff
@@ -45,6 +45,55 @@
...
+    display: inline;
+}
+
+.profile-welcome .profile-curated-card-description .profile-welcome-shimmer {
+    height: 32px;
+    color: transparent;
+    display: inline;
+}
+
+.profile-welcome .profile-user-card-avatar:has(.profile-welcome-shimmer) {
```
Comment (nitin-rachabathuni) [inline blocks/profile-welcome/profile-welcome.css:93]: Refer transcriptLoading

File: blocks/profile-welcome/profile-welcome.js [hunk 3]
```diff
@@ -37,27 +37,45 @@ async function decorateProfileWelcomeBlock(block) {
...
 
-  const {
-    displayName: adobeDisplayName = UEAuthorMode ? 'User Name' : '',
-    first_name: adobeFirstName = UEAuthorMode ? 'First Name' : '',
-    industryInterests: industry = UEAuthorMode ? ['User Industry'] : [],
-    role: roles = UEAuthorMode ? ['User Roles'] : [],
-    interests = UEAuthorMode ? ['User Interests'] : [],
-  } = profileData || {};
+  const getProfileInfoData = async () => {
+    const { profileData, ppsProfileData } = await profilePromise;
```
Comment (nitin-rachabathuni) [inline blocks/profile-welcome/profile-welcome.js:59]: rename fetchProfileData

File: blocks/profile-welcome/profile-welcome.js [hunk 4]
```diff
@@ -124,75 +138,95 @@ async function decorateProfileWelcomeBlock(block) {
...
                     <span class="role-heading">${placeholders?.myRole || 'MY ROLE'}: </span>
-                    <span class="${!hasInterests ? 'incomplete-profile' : 'content'}">${roles
-                      ?.map((role) => roleMappings[role] || role)
-                      ?.join(' | ')}</span>
+                    <span class="profile-role-contents">
+                      <p class="loading-shimmer" style="--placeholder-width: 100%; height: 32px"></p>
+                    </span>
+                    </div>
+                    <div class="profile-user-card-industry">
+                      <p class="loading-shimmer" style="--placeholder-width: 100%; height: 32px"></p>
```
Comment (nitin-rachabathuni) [inline blocks/profile-welcome/profile-welcome.js:224]: Try to use --height attribute

File: blocks/profile-welcome/profile-welcome.js [hunk 5]
```diff
@@ -202,15 +236,92 @@ async function decorateProfileWelcomeBlock(block) {
...
 export default async function decorate(block) {
   const blockInnerHTML = block.innerHTML;
-  await decorateProfileWelcomeBlock(block);
-
-  profileEventEmitter.on('profileDataUpdated', async () => {
-    block.innerHTML = blockInnerHTML;
-    await decorateProfileWelcomeBlock(block);
+  decorateProfileWelcomeBlock(block).then(() => {
+    profileEventEmitter.on('profileDataUpdated', async () => {
+      block.innerHTML = blockInnerHTML;
```
Comment (nitin-rachabathuni) [inline blocks/profile-welcome/profile-welcome.js:323]: Check sign-up pop-up flow.
Open, close pop-up, ensure shimmers are still visible as php is reloaded in this case.

**Review by gauthamksn — APPROVED** [review]

**Review by zeegeek — COMMENTED** [review]
C

**Comment by noormd08** [comment]
Changes looks fine and reviewed the page as well. Any additional fixes to be considered as separate PR. Please make sure to test the complete functionality before releasing it to QA.

**Pull request merged**